packages/xc-sidebar/src/styled/Sidebar.tsx

```diff
@@ -1,110 +1,109 @@
 import styled from 'styled-components';
-
-
-interface ISidebarTheme {
-  leftBackground: string;
-  rightBackground: string;
-  color: string;
-  separatorColor: string;
-  minWidth: number;
-  maxWidth: number;
-  leftWidth: number;
-}
-
-interface IProps {
-  theme: ISidebarTheme;
-}
-
-
-interface IResponsiveWrapper {
-  animate: boolean;
-}
-interface IChildWrapper extends IProps {
-  animate: boolean;
-}
-
-
-interface ICloseOpenButton {
-  toRight: boolean;
-}
-
-
-export const RightBorder = styled.div<IProps>`
-  background-color: ${({ theme: { separatorColor } }: IProps): string => separatorColor};
-  width: 2px;
-  height: 100% ;
-  float: right;
-  position: relative;
-  right: 10px;
-
-`;
-export const SidebarWrapper = styled.div<IProps>`
-  color: ${({ theme: { color } }: IProps): string => color};
-  height: 100vh ;
-  position: fixed;
-  top: 0;
-  left: 0;
+import { Scrollbars } from 'react-custom-scrollbars';
+
+import {
+  getBaseStyle,
+  getWidthStyles,
+  getPropertyStyles,
+  getBackgroundColorStyles,
+  getElementStyles,
+} from '../utils';
+import {
+  ISidebarProps,
+  ISidebarStates,
+  IResponsiveWrapper,
+  IChildWrapper,
+  ICloseOpenButton,
+  IScrollbarProps,
+} from '../interfaces';
+
+
+export const Root = styled.div.attrs(({ offsetLeft }: ISidebarStates) => ({
+  style: {
+    width: `${offsetLeft}px`,
+  },
+}))<ISidebarProps & ISidebarStates>`
+  ${({ theme, appearance, baseAppearance }) => getBaseStyle(theme, appearance, baseAppearance)};
+  float: ${({ isRTL }) => (isRTL ? 'right' : 'left')};
 `;
 
-export const NavComponentWrapper = styled.div<IProps>`
-  background-color: ${({ theme: { leftBackground } }: IProps): string => leftBackground};
-  height: calc(100vh + 10px);
-  float: left;
+export const SidebarWrapper = styled.div<ISidebarProps>`
+  ${({ theme, appearance, baseAppearance }) => getPropertyStyles(theme, 'color', appearance, baseAppearance)('sidebarContainer')};
```

Comment:
`${({ theme, appearance, baseAppearance }) => sidebarAppearanceTheme(theme, appearance, baseAppearance, 'sidebarContainer');`

Instead of doing it for single properties, do it for all.
So i should be able pass any cssProp that i want (i need zIndex for example).